test/unit/model/PermissionsServiceTest.php

```diff
@@ -245,7 +244,7 @@ public function testDoNotSaveAnythingWhenThereIsNothingToSave(): void
         $this->databaseAccess->expects($this->never())->method('addPermissions');
         $this->databaseAccess->expects($this->never())->method('removePermissions');
 
-        $this->databaseAccess->method('getResourcePermissions')->willReturn([]);
+        $this->databaseAccess->method('getResourcesPermissions')->willReturn([]);
```

Comment:
The correct method name is actually `getResourcesPermissions` (with "resources" in plural), see https://github.com/oat-sa/extension-tao-dac-simple/blob/feat/ADF-1320/prototype-subtasks-acl/model/DataBaseAccess.php#L129

Reply:
There is also a method `getResourcePermissions`, that exists, but was removed in your refactoring.

Reply:
This PR is not making any changes in the `DatabaseAccess` class, therefore this test was already failing before.

The method this test refers to is here: https://github.com/oat-sa/extension-tao-dac-simple/blob/feat/ADF-1320/prototype-subtasks-acl/model/DataBaseAccess.php#L129

Reply:
Ok, got it: The call was changed, and now `PermissionsService` always calls the version in plural.

Reply:
Please make sure that this change did not cause any bug as the methods have different response format :)

Reply:
It is working as expected. Previously we called the singular form to get the permissions for the root, computed the delta based on them (calling the strategy), and did an early return if the delta was empty. In the delta was not empty, we delegated in the strategy to compute the permissions to change for each class or resource contained in the root.

Now we use `getDeltaForResourceTree()` to get a diff for each resource (again, using the strategy), and exit early if there is no change to perform for **any** resource. To do that, we need to get the permissions for each resource first (i.e. the method in plural).

That allows us to propagate an ACL change in a class (say, "add READ for U1 in C1") to a resource R (that does not have READ for U1) in a nested class C2  even if C2 already had the READ permission for U1.

```
C1 --> We add READ here
-- C2 --> Already has READ access 
---- R --> Does not have READ access
```

If we don't check the permissions for individual resources (method in plural) we would not be able to propagate the change as soon as we encounter a class that already has that permission.